torch/distributed/_sharded_tensor/api.py

```diff
@@ -123,6 +125,26 @@ def _register_remote_shards(sharded_tensor_id: int, rrefs: List[rpc.RRef[Shard]]
         _sharded_tensor_map[sharded_tensor_id]._register_remote_shards(rrefs, rpc_rank)
 
 
+class CreateOp(Enum):
+    EMPTY = 0
+    ONES = 1
+
+
+@dataclass
+class TensorInitParams(object):
+    """ Container for list of common params to create new local tensor. """
+
+    __slots__ = ['create_op', 'dtype', 'layout', 'requires_grad', 'pin_memory',
+                 'memory_format']
+
+    create_op: CreateOp
+    dtype: torch.dtype
```

Comment:
nit: just a minor suggestion and can be in a follow up diff. I noticed that these fields have a big overlap with ShardedTensorMetadata, maybe we can group these tensor relevant properties together as TensorProperties (i.e. dtype, layout, requires_grad, pin_memory, memory_format), then ShardedTensorMetadata could just take that and a list of metadatas, sharded tensor constructors can just take `create_op` and tensor property object.

Reply:
Great suggestion Wanchao, will send out a follow up PR for this.

Reply:
Follow up: sent out the consolidation PR at https://github.com/pytorch/pytorch/pull/63731